auto-generated/api-testing/vlseg2e8.c

```diff
@@ -8,95 +8,191 @@ void test_vlseg2e8_v_i8mf8(vint8mf8_t *v0, vint8mf8_t *v1, const int8_t *base, s
   return __riscv_vlseg2e8_v_i8mf8(v0, v1, base, vl);
 }
 
+void test_vlseg2e8_v_i8mf8(vint8mf8x2_t *v_tuple, const int8_t *base, size_t vl) {
```

Comment:
It looks weird that make the `void` to be the return value from the memory load operation's perspective.
* It is not consistent with other loads which return the loaded value itself.
* The inconsistency may bring kind of surprise from the users' perspective.
* Return value with parameters is one well-known anti-pattern unless you have no other choose.

Thus, how about just return the loaded structure directly and natively?

Reply:
> * It is not consistent with other loads which return the loaded value itself.
> * The inconsistency may bring kind of surprise from the users' perspective.
> * Return value with parameters is one well-known anti-pattern unless you have no other choose.

I disagree with this.

When the return value is larger than 2 * XLEN bits, it requires a temporary memory allocation on callee's stack to hold the return value, then returning the pointer points to the temporary stack location instead. Accessing callee's stack after callee is exited is a danger behavior, user must make sure no following function calls happen before pull out data from callee's stack.

Instead, we should allocate memory at caller stack and pass it as function argument to the callee(intrinsic) to pass back the loaded value.

Reply:
If RVV intrinsics are following the "standard" calling convention, we are in trouble...
https://github.com/riscv-non-isa/riscv-elf-psabi-doc/blob/master/riscv-cc.adoc#vector-register-convention

Reply:
> If RVV intrinsics are following the "standard" calling convention, we are in trouble...

Agree, I think it's better to have vector register calling convention be defined to pass in/back function arguments and return value.

Reply:
Intrinsic does not need to follow calling convention, it's builtin function and will expand/inline during compilation.


And I am support that should just use tuple type as return value instead of pass by pointer.

Reply:
> Intrinsic does not need to follow calling convention, it's builtin function and will expand/inline during compilation.
> 
> And I am support that should just use tuple type as return value instead of pass by pointer.

Ops, missed this one.